File: MaratoneiA1/app/src/main/java/com/example/manoel/maratoneia1/WelcomeActivity.java
package com.example.manoel.maratoneia1;

import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.dataMovie.manoel.maratoneia1.R;

public class WelcomeActivity extends AppCompatActivity implements View.OnClickListener {

    private Button btnEnter;
    private static final String ARQUIVO_PREFERENCIA = "FileReference";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SharedPreferences preferences = getSharedPreferences(ARQUIVO_PREFERENCIA,MODE_PRIVATE);
        if(preferences.contains("enter")){
            Intent intent = new Intent(this,MainActivity.class);
            startActivity(intent);
            this.finish();
        }
        else{
            setContentView(R.layout.activity_welcome);
            btnEnter = findViewById(R.id.btnAccess);
            btnEnter.setOnClickListener(this);
        }
    }

    @Override
    public void onClick(View v) {
        SharedPreferences preferences = getSharedPreferences(ARQUIVO_PREFERENCIA,MODE_PRIVATE);
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean("enter", true);
        editor.commit();
        Intent intent = new Intent(this,MainActivity.class);
        startActivity(intent);
        this.finish();
    }
}

File: MaratoneiKotlin/app/src/main/java/com/example/manoeltoj/maratoneikotlin/WelcomeActivity.kt
package com.example.manoeltoj.maratoneikotlin

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button

class WelcomeActivity: AppCompatActivity(),View.OnClickListener{


    lateinit var btnEnter: Button

    companion object {
        val ARQUIVO_PREFERENCIA:String = "ArquivoPreferencia"
    }
    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        var preferences:SharedPreferences = getSharedPreferences(ARQUIVO_PREFERENCIA,MODE_PRIVATE)
        if(preferences.contains("intro")){
            var intent:Intent =  Intent(this, MainActivity::class.java)
            startActivity(intent)
            this.finish()
        }else{
            setContentView(R.layout.activity_welcome)
            btnEnter = findViewById(R.id.btnAccess)
            btnEnter.setOnClickListener(this)
        }

        setContentView(R.layout.activity_welcome)
    }

    override fun onClick(v: View?) {
        var preferences:SharedPreferences = getSharedPreferences(ARQUIVO_PREFERENCIA,MODE_PRIVATE)
        var editor:SharedPreferences.Editor = preferences.edit()
        editor.putBoolean("intro",true)
        editor.commit()
        var intent:Intent = Intent(this,MainActivity::class.java)
        startActivity(intent)
        this.finish()
    }

}
